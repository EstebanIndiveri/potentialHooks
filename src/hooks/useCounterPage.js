import React, { useState } from 'react';

const useCounterPage = (initalState=10) => {
    const[counter,setState]=useState(initalState);
    const increment=()=>{
        setState(counter+1);
    }
    const decrement=()=>{
        setState(counter-1);
    }
    const reset=()=>{
        setState(counter);
    }
    return{
        counter,
        increment,
        decrement,
        reset
    };
}
 
export default useCounterPage;