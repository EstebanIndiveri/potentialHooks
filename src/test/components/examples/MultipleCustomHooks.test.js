import React from 'react';
import MultipleCustomHooks from '../../../components/examples/MultipleCustomHooks'
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import useFetch from '../../../hooks/useFetch';
import useCounterPage from '../../../hooks/useCounterPage';
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounterPage');

describe('multiple custom hooks', () => {

    useCounterPage.mockReturnValue({
        counter:10,
        increment:()=>{}
    });

    test('debe mostrar correctamente', () => {
        useFetch.mockReturnValue({
            data:null,
            loading:true,
            error:null
        })
        const wrapper=shallow(<MultipleCustomHooks/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar info', () => {
        useFetch.mockReturnValue({
            data:[{
                author:'fer',
                quote:'hola mundo'
            }],
            loading:false,
            error:null
        });
        const wrapper=shallow(<MultipleCustomHooks/>)

        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('hola mundo');
        expect(wrapper.find('footer').text().trim()).toBe('fer');


    })
    
    
    
})
