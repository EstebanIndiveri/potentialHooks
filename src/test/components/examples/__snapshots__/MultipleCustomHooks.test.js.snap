// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multiple custom hooks debe mostrar correctamente 1`] = `
<Fragment>
  <div>
    BreakingBad Quotes
  </div>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Siguiente quote
  </button>
</Fragment>
`;
